oat-sa/extension-tao-system-status
# Feature/lead 86/opsgenie alerts

Implementation of opsgenie alert service.

How to test:
1. register service `config/tao/alarmNotificationService.conf.php`: 
```
<?php
return new oat\tao\model\notifications\AlarmNotificationService(array(
    'notifiers' => array(
        [
            'class' => '\\oat\\tao\\model\\notifiers\\OpsGenieNotifier',
            'params' => ['<api_key>']
        ]
    )
));
```
(you can get api key here: https://oat-sa.app.eu.opsgenie.com/settings/integration/edit/API/6aee2a6b-910a-4ebc-9fbb-6b34992f6d03?teamId=fcbac5df-de27-4818-8b42-feb40d078f26 or contact me in DM)

2. send an alert:
```
sudo -u www-data php index.php '\oat\taoSystemStatus\scripts\tools\CreateAlert' -m '<alert message>' -d '<alert description>'
```

3. Go to the alerts list page: https://oat-sa.app.eu.opsgenie.com/alert
New alert with appropriate message/description should be created.

4. Run `php index.php '\oat\taoSystemStatus\scripts\tools\CheckDegradations' -p default_kv`

5. change namespace option in `config/generis/uriProvider.conf.php` in order to make `LocalNamespaceCheck` fail

6. Run `php index.php '\oat\taoSystemStatus\scripts\tools\CheckDegradations' -p default_kv` again. Since a degradation found new alert with degradation description must be created.
